manifests/init.pp

```diff
@@ -61,7 +61,33 @@
   Optional[Integer] $logconfig_file_max_log_size = undef,
   Optional[Integer] $logconfig_file_backups      = undef,
   Optional[String] $logconfig_file_category      = undef,
-) {
+
+  # Padoptions
+  #Optional[Hash[String[0],String[1]]] $padoptions = undef,
+  Optional[Hash[Enum['nocolors',
```

Comment:
I think you can leave out the `Optional` and default to an empty hash. Merging should still work with that.

Reply:
Sorry but if I leave out the ```optional```, if a user put a wrong word (like a word without "s" or with a capital letter), he couldn't be able to see were is the error. I think it's more safe like that.  

What do you think about this ?

Reply:
I'm not sure how `Optional[Hash[...]]` provides that safety but `Hash[...]` doesn't.

I'm also wondering if we should use a struct like https://github.com/camptocamp/puppet-systemd/blob/master/types/servicelimits.pp to keep this file readable.

Reply:
`Optional` just means that `undef` is a valid value for `$padoptions`.  @ekohl is suggesting removing the `Optional` and changing the default from `undef` (which would no longer be valid) to an empty hash `{}`